### PR TITLE
Refactoring flaky test suites to run reliably

### PR DESCRIPTION
**Pull request by meenakb**

### CODE REVIEW

File: src/taco-cli/test/kit.ts [hunk 1]
```diff
@@ -214,67 +213,67 @@ describe("Kit command : ", function (): void {
...
-                telemetryParameters.should.be.eql(expected);
-                done();
-            }, function (err: tacoUtils.TacoError): void {
-                done(err);
-            });
-    });
+    describe("Kit list command : ", function (): void {
+        it("'taco kit' should not throw any error", function (done: MochaDone): void {
+            kitRun()
+                .done((telemetryParameters: TacoUtility.ICommandTelemetryProperties) => {
```
Comment (MSLaguana) [inline src/taco-cli/test/kit.ts]: Looks like we don't need the done handler, so we can change this to just return kitRun().then(...)

Comment (meenakb) [inline src/taco-cli/test/kit.ts]: Makes sense. @nisheetjain, can you review this?

File: src/taco-cli/test/kit.ts [hunk 2]
```diff
@@ -214,67 +213,67 @@ describe("Kit command : ", function (): void {
...
         rimraf.sync(runFolder);
     });
 
-    after((done: MochaDone) => {
+    after() => {
         process.env["TACO_UNIT_TEST"] = previous;
         process.chdir(originalCwd);
         kitHelper.kitPackagePromise = null;
-        rimraf(runFolder, function (err: Error): void { done(); }); // ignore errors
+        rimraf.sync(runFolder); // ignore errors
```
Comment (MSLaguana) [inline src/taco-cli/test/kit.ts]: Is this the change that resolved the timeouts?

Comment (meenakb) [inline src/taco-cli/test/kit.ts]: Correct :)

**Pull request merged**